exonum/src/blockchain/mod.rs

```diff
@@ -234,7 +234,10 @@ impl BlockchainMut {
                 instance_config.constructor,
             )?;
         }
-        self.dispatcher.after_commit(&fork);
+        // We need to activate services before calling `create_patch()`; unlike all other blocks,
+        // initial services are considered immediately active in the genesis block, i.e.,
+        // their state should be included into `patch` created below.
+        self.dispatcher.commit_block(&fork);
```

Comment:
IMO, this split is sort of cringe, but trying to apply usual rules (services are activated _after_ the block is committed) leads to some problems:

- Lots of broken tests (e.g., in the sandbox)
- More importantly, with this change the hash of the genesis block would not depend on the included services! This brings us back to the genesis block having real transactions (as in: transactions to deploy some artifacts and initialize some services).